### PR TITLE
[FIX] VRecDialog Camera Permission Issue Resolved

### LINKED ISSUE #24279: [BUG] VRecDialog Opening Also, If Permission Is Set To Block Camera
### Description:
If the user manually changes the camera permission from allowing to block, We should not open VRecDialog at that time.

### Steps to reproduce:

1. Go to room
2. Click on  the **+** icon in the message box section
3. Click on Video Message.
4. If the user blocked the camera permission, then It will open blank VRecDailog.

### Expected behavior:
If the camera permission is allowed then it's ok to show VRecDialog, But when the camera is blocked, we should not open VRecDialog and send a message to change the camera permission setting.

### Actual behavior:

If camera permission is blocked, Still it opens a blank VRecDialog.

https://user-images.githubusercontent.com/53515714/150952590-9a0d1b3e-e085-44c9-a5ee-186a452c6743.mp4

### Server Setup Information:

- Version of Rocket.Chat Server: 4.4.0
- Operating System: Windows 10
- Deployment Method: 
- Number of Running Instances: 
- DB Replicaset Oplog: 
- NodeJS Version: 14.18.2
- MongoDB Version: 4.4.4 

### Client Setup Information

- Desktop App or Browser Version:
- Operating System:

### PR DESCRIPTION
**Pull request by nishant23122000**
## Proposed changes (including videos or screenshots)

Mention in issue #24279

After resolving thee bug:

https://user-images.githubusercontent.com/53515714/150957806-99c1c7cf-bc99-4619-9e5e-fbbe678c1225.mp4

## Issue(s)
Closes #24279

## Steps to test or reproduce
available in issue number #24279

### CODE REVIEW

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 1]
```diff
@@ -713,6 +713,7 @@
   "call-management": "Call Management",
   "call-management_description": "Permission to start a meeting",
   "Caller": "Caller",
+  "Camera_is_blocked":"Camers is blocked, change the browser setting to allow it",
```
Comment (debdutdeb) [inline packages/rocketchat-i18n/i18n/en.i18n.json:716]:
```suggestion
  "Camera_is_blocked": "Camera is blocked, change the browser setting to allow it",
```

**Review by debdutdeb — CHANGES_REQUESTED** [review]

**Comment by nishant23122000** [comment]
Hi @murtaza98  Can you merge this PR?

**Review by murtaza98 — APPROVED** [review]
LGTM!

**Comment by murtaza98** [comment]
Hi @nishant23122000 Some tests are failing here. Do u mind taking a look?

**Comment by nishant23122000** [comment]
Ok let me check

**Comment by nishant23122000** [comment]
Hi, @murtaza98 Please check now.

File: apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json [hunk 2]
```diff
@@ -728,6 +728,7 @@
   "call-management_description": "Permission to start a meeting",
   "Caller": "Caller",
   "Caller_Id": "Caller ID",
+  "Camera_is_blocked":"Camers is blocked, change the browser setting to allow it",
```
Comment (dougfabris) [inline apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json:731]:
```suggestion
  "Camera_blocked_change_browser_setting_to_allow": "Camera is blocked, change the browser setting to allow it",
```

File: apps/meteor/app/ui-message/client/messageBox/messageBoxActions.js [hunk 3]
```diff
@@ -23,7 +24,11 @@ messageBox.actions.add('Create_new', 'Video_message', {
 		settings.get('Message_VideoRecorderEnabled') &&
 		(!settings.get('FileUpload_MediaTypeBlackList') || !settings.get('FileUpload_MediaTypeBlackList').match(/video\/webm|video\/\*/i)) &&
 		(!settings.get('FileUpload_MediaTypeWhiteList') || settings.get('FileUpload_MediaTypeWhiteList').match(/video\/webm|video\/\*/i)),
-	action: ({ rid, tmid, messageBox }) => (VRecDialog.opened ? VRecDialog.close() : VRecDialog.open(messageBox, { rid, tmid })),
+	action: async ({ rid, tmid, messageBox }) => {
+		const permissionStatus = await navigator.permissions.query({ name: 'camera' });
+		if (permissionStatus.state === 'denied') return dispatchToastMessage({ type: 'error', message: t('Camera_is_blocked') });
```
Comment (dougfabris) [inline apps/meteor/app/ui-message/client/messageBox/messageBoxActions.js:29]:
```suggestion
		if (permissionStatus.state === 'denied') return dispatchToastMessage({ type: 'error', message: t('Camera_blocked_change_browser_setting_to_allow') });
```

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json (hunk 2 above)
Comment (nishant23122000) [reply apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json]: Done

File: apps/meteor/app/ui-message/client/messageBox/messageBoxActions.js (hunk 3 above)
Comment (nishant23122000) [reply apps/meteor/app/ui-message/client/messageBox/messageBoxActions.js]: Hey @dougfabris Done, Thanks

**Comment by nishant23122000** [comment]
check now 🙂

**Comment by murtaza98** [comment]
Thanks, @nishant23122000. Please confirm if this PR is still relevant in light of the most recent modifications to message composer. If so, kindly assist us in resolving the conflicts so that this PR can be moved for review once more :pray:

**Pull request closed without merging**